test/e2e-external/size_test.go

```diff
@@ -0,0 +1,130 @@
+package e2e
+
+import (
+	"context"
+	mmv1alpha "github.com/mattermost/mattermost-operator/apis/mattermost/v1alpha1"
+	mmv1beta "github.com/mattermost/mattermost-operator/apis/mattermost/v1beta1"
+	"github.com/mattermost/mattermost-operator/test/e2e"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"testing"
+	"time"
+)
+
+// TestMattermostSize checks defaulting & updating replicas & resources from size.
+func TestMattermostSize(t *testing.T) {
+	namespace := "e2e-test-size"
+	name := "test-mm"
+	mmNamespaceName := types.NamespacedName{Namespace: namespace, Name: name}
+
+	testEnv, setupErr := SetupTestEnv(k8sClient, namespace)
+	require.NoError(t, setupErr)
+	defer testEnv.CleanupFunc()
+
+	mattermost := &mmv1beta.Mattermost{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: mmv1beta.MattermostSpec{
+			Ingress: &mmv1beta.Ingress{
+				Host: "e2e-test-example.mattermost.dev",
+			},
+			FileStore: mmv1beta.FileStore{
+				External: &testEnv.FileStoreConfig,
+			},
+			Database: mmv1beta.Database{
+				External: &testEnv.DBConfig,
+			},
+		},
+	}
+
+	{
```

Comment:
Any specific reason for enclosing it in additional braces?

Reply:
This enclosure is used to make this test easier to read. This contains code blocks that perform a certain action of the test.

Reply:
Instead of just braces, could you change them to Go subtests? That way we also have nice headers and test time data.

Reply:
That is a great idea.  Resolved with Go subtests.